app/vmauth/main.go

```diff
@@ -103,7 +99,13 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 	for _, h := range headers {
 		r.Header.Set(h.Name, h.Value)
 	}
-	proxyRequest(w, r)
+
+	requester := func() { proxyRequest(w, r) }
+	if err := ui.proxyRequests(requester); err != nil {
+		errStr := fmt.Sprintf("cannot proxy request: %s", err)
+		http.Error(w, errStr, http.StatusTooManyRequests)
```

Comment:
retry-after header is missing https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429

Reply:
added retry-after header